model/jpa/src/main/java/org/jboss/aerogear/unifiedpush/jpa/dao/impl/JPABaseDao.java

```diff
@@ -62,6 +62,12 @@ public void update(T entity) {
 
     public void delete(T entity) {
         if (entity != null) {
+            // making sure the entity in question,
+            // is really part of this transaction
+            if (! entityManager.contains(entity)) {
+                entity = entityManager.merge(entity);
```

Comment:
@edewit I have no info on the .getId() - instead of casting I did this.

However, I can see this a nice improvement. OK if we file a JIRA for 1.1.x?